### PR TITLE
Format array literals onto a single line if no trailing comma

### PR DESCRIPTION
**Pull request by LewisGaul**
Fixes https://github.com/ziglang/zig/issues/5176

Includes change in behaviour (see tweaked testcases): unless an array literal contains a comment or multiline string, there must be a trailing comma for it to be formatted over multiple lines.

cc @ifreund

### CODE REVIEW

File: lib/std/zig/parser_test.zig [hunk 1]
```diff
@@ -1801,19 +1801,19 @@ test "zig fmt: array literal with hint" {
...
+        \\    50000, 60,
         \\};
         \\const a = []u8{0,   1, 2, 3, 40,
         \\    4,5,600,7,
         \\           80,
-        \\    9, 10, 11, 0, 13, 14, 15};
+        \\    9, 10, 11, 0, 13, 14, 15,};
         \\const a = [12]u8{
-        \\    31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
+        \\    31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31, };
```
Comment (ifreund) [inline lib/std/zig/parser_test.zig:1816]: could we not change this test case? it was added to catch this bug: https://github.com/ziglang/zig/issues/8423

**Review by ifreund — COMMENTED** [review]
LGTM!

File: lib/std/zig/parser_test.zig (hunk 1 above)
Comment (LewisGaul) [reply lib/std/zig/parser_test.zig:1818]: I had to add the tailing comma for the same code path to be exercised, but I've copied the same case minus the trailing comma anyway.

**Review by ifreund — APPROVED** [review]
Thanks!

**Pull request merged**